src/block/block_ext.c

```diff
@@ -622,6 +622,9 @@ __wt_block_off_free(
 {
 	WT_DECL_RET;
 
+	WT_ASSERT(session, S2BT(session)->checkpointing != WT_CKPT_RUNNING ||
+	   WT_SESSION_TREE_SYNC(session));
+
```

Comment:
This is the check I used to track down when things were going wrong: ideally I'd like to catch the general case without a layering violation.

Reply:
I don't feel strongly about this kind of assert in the default block manager, it's good with me. Would it be worth adding a more general assertion if a block is allocated or freed in a file that's being checkpointed?

Reply:
Trivial white-space error, the follow-on line is incorrectly indented.